src/wp-includes/option.php

```diff
@@ -3214,3 +3214,126 @@ function wp_autoload_values_to_autoload() {
 
 	return array_intersect( $filtered_values, $autoload_values );
 }
+
+
+/**
+ * Filters site metadata for multisite installations.
+ *
+ * This function is specific to multisite setups and provides a way to retrieve
+ * network options. It enforces parameters' constraints and suggests using an alternative function.
+ *
+ * @param mixed $current The value to return if no changes are made.
+ * @param int $object_id Site ID.
+ * @param string $meta_key Metadata key.
+ * @param bool $single Whether to retrieve a single value. Must be true.
+ *
+ * @return mixed The network option value for the specified meta key, or false if the function is used incorrectly.
+ */
+function filter_get_site_metadata( $current, $object_id, $meta_key, $single ) {
+	if ( ! is_multisite() ) {
+		_doing_it_wrong( 'get_metadata', __( 'This function is only available in multisite installations.' ), '6.8.0' );
+		return false;
+	}
+
+	if ( ! $single ) {
+		_doing_it_wrong( 'get_metadata', __( 'The $single parameter is not supported in this function.' ), '6.8.0' );
+		return $current;
+	}
+
+	_doing_it_wrong( 'get_metadata', __( 'This function is deprecated. Use get_network_option() instead.' ), '6.8.0' );
+
+	return get_network_option( $object_id, $meta_key );
+}
+
+/**
+ * Filters and updates site metadata in a multisite environment.
+ *
+ * This function is only applicable in multisite installations. It has been
+ * deprecated in favor of using `update_network_option()`.
+ *
+ * @param mixed $current The current metadata value. Unused.
+ * @param int $object_id Site ID for which the metadata is being updated.
+ * @param string $meta_key Metadata key to update.
+ * @param mixed $meta_value Metadata value to update.
+ *
+ * @return bool|void False if the operation is not applicable or invalid,
+ *                   or the result of `update_network_option()` on success.
+ */
+function filter_update_site_metadata( $current, $object_id, $meta_key, $meta_value ) {
+	if ( ! is_multisite() ) {
+		_doing_it_wrong( 'update_metadata', __( 'This function is only available in multisite installations.' ), '6.8.0' );
+		return false;
+	}
+
+	_doing_it_wrong( 'update_metadata', __( 'This function is deprecated. Use update_network_option() instead.' ), '6.8.0' );
+
+	return update_network_option( $object_id, $meta_key, $meta_value );
```

Comment:
See above, I think this could simply become:
```suggestion
	return update_network_option( $object_id, $meta_key, $meta_value );
```

Reply:
I'm more for being loud and expressive. So these `doing_it_wrong`s are fine for me. :-)

Reply:
I think they are excessive, as both can be made to work. But more importantly, some of the messages are confusing. For example `update_metadata()` is not a deprecated function.